### ISSUE TITLE
binlog replication: wrong location compare if not enable GTID

### ISSUE THREAD

**Issue by csuzhangxc** [pingcap/dm#584]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

   two MySQL instances in upstream with optimistic mode, add a column with different default value in upstream.

    for MySQL-1.
    ```sql
    alter table tbl add column c2 varchar(100) default '1234';
    ```

    for MySQL-2.
    ```sql
    alter table tbl add column c2 varchar(100) default '5678';
    ```

2. What did you expect to see?

    block DDL for the table in MySQL-2 even auto resume trigger.

3. What did you see instead?

    DDL for the table in MySQL-2 ignored after auto resume.

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        nightly before April 8.
        ```

```sql
| Log_name         | Pos  | Event_type     | Server_id | End_log_pos | Info                                                                |
+------------------+------+----------------+-----------+-------------+---------------------------------------------------------------------+
| mysql-bin.000003 | 4    | Format_desc    | 12        | 123         | Server ver: 5.7.22-log, Binlog ver: 4                               |
| mysql-bin.000003 | 123  | Previous_gtids | 12        | 194         | 11ee15b1-796d-11ea-85ec-0242ac130003:1-5                            |
| mysql-bin.000003 | 194  | Gtid           | 12        | 259         | SET @@SESSION.GTID_NEXT= '11ee15b1-796d-11ea-85ec-0242ac130003:6'   |
| mysql-bin.000003 | 259  | Query          | 12        | 347         | create database db                                                  |
| mysql-bin.000003 | 347  | Gtid           | 12        | 412         | SET @@SESSION.GTID_NEXT= '11ee15b1-796d-11ea-85ec-0242ac130003:7'   |
| mysql-bin.000003 | 412  | Query          | 12        | 519         | use `db`; create table tbl (c1 int primary key)                     |
| mysql-bin.000003 | 519  | Gtid           | 12        | 584         | SET @@SESSION.GTID_NEXT= '11ee15b1-796d-11ea-85ec-0242ac130003:8'   |
| mysql-bin.000003 | 584  | Query          | 12        | 654         | BEGIN                                                               |
| mysql-bin.000003 | 654  | Table_map      | 12        | 698         | table_id: 110 (db.tbl)                                              |
| mysql-bin.000003 | 698  | Write_rows     | 12        | 738         | table_id: 110 flags: STMT_END_F                                     |
| mysql-bin.000003 | 738  | Xid            | 12        | 769         | COMMIT /* xid=38 */                                                 |
| mysql-bin.000003 | 769  | Gtid           | 12        | 834         | SET @@SESSION.GTID_NEXT= '11ee15b1-796d-11ea-85ec-0242ac130003:9'   |
| mysql-bin.000003 | 834  | Query          | 12        | 904         | BEGIN                                                               |
| mysql-bin.000003 | 904  | Table_map      | 12        | 948         | table_id: 220 (db.tbl)                                              |
| mysql-bin.000003 | 948  | Write_rows     | 12        | 988         | table_id: 220 flags: STMT_END_F                                     |
| mysql-bin.000003 | 988  | Xid            | 12        | 1019        | COMMIT /* xid=109 */                                                |
| mysql-bin.000003 | 1019 | Gtid           | 12        | 1084        | SET @@SESSION.GTID_NEXT= '11ee15b1-796d-11ea-85ec-0242ac130003:10'  |
| mysql-bin.000003 | 1084 | Query          | 12        | 1211        | use `db`; alter table tbl add column c2 varchar(100) default '5678' |
| mysql-bin.000003 | 1211 | Gtid           | 12        | 1276        | SET @@SESSION.GTID_NEXT= '11ee15b1-796d-11ea-85ec-0242ac130003:11'  |
| mysql-bin.000003 | 1276 | Query          | 12        | 1346        | BEGIN                                                               |
| mysql-bin.000003 | 1346 | Table_map      | 12        | 1393        | table_id: 222 (db.tbl)                                              |
| mysql-bin.000003 | 1393 | Write_rows     | 12        | 1438        | table_id: 222 flags: STMT_END_F                                     |
| mysql-bin.000003 | 1438 | Xid            | 12        | 1469        | COMMIT /* xid=177 */                                                |
+------------------+------+----------------+-----------+-------------+---------------------------------------------------------------------+
```

[dm-worker.log](https://github.com/pingcap/dm/files/4449291/dm-worker.log)

**Comment by WangXiangUSTC** [comment]
fix in syncer #614

**Comment by csuzhangxc** [comment]
#625 also fix a bug relative to this issue.
